apps/vessel-history/features/home/Home.tsx

```diff
@@ -96,26 +96,37 @@ const Home: React.FC<LoaderProps> = (): React.ReactElement => {
     },
     [dispatch]
   )
+
+  const getListOfSelectedVessels = useCallback(() => {
+    return selectedVessels
+      .map((index) => vessels[index])
+      .map((allVessels) =>
+        allVessels.relatedVessels
+      ).flatMap(relatedVessel => relatedVessel)
+  }, [vessels, selectedVessels])
+
+
   const onOpenVesselProfile = useCallback(
     (vessel) => () => openVesselProfile(vessel),
     [openVesselProfile]
   )
 
   const onSeeVesselClick = useCallback(() => {
-    const selectedVessel = vessels[selectedVessels[0]]
+    const parsedSelectedVessels = getListOfSelectedVessels()
+    const selectedVessel = parsedSelectedVessels[0]
     if (selectedVessel) openVesselProfile(selectedVessel)
   }, [openVesselProfile, selectedVessels, vessels])
 
   const onMergeVesselClick = useCallback(() => {
```

Comment:
This is never executed now when the user only selects an auto-merged vessel because on lines 316:327 of this same file we are displaying `SEE VESSEL` button when there is only one vessel selected (not MERGE SELECTED VESSELS button). I'd leave only the `SEE VESSEL` button there and inside the onSeeVesselClick open the profile of a single vessel when only one vessel without related vessels is selected, but if multiple vessels are selected or one vessel with related vessels then open the profile for all of them. Take into account that browsers do have a maximum length for urls (https://www.geeksforgeeks.org/maximum-length-of-a-url-in-different-browsers/)

Reply:
no, both buttons are used, so this function is used too